core/domain/entities/editor/Block.php

```diff
@@ -181,21 +173,19 @@ public function setDynamic($dynamic = true)
      */
     public function registerBlock()
     {
-        $context ='core';
-        // todo add route detection (ie inject route) and change context based on route
         $args          = array(
             'attributes'      => $this->attributes(),
-            'editor_script'   => "ee-{$context}-blocks",
-            'editor_style'    => "ee-{$context}-blocks",
-            'script'          => "ee-{$context}-blocks",
-            'style'           => "ee-{$context}-blocks",
+            'editor_script'   => $this->asset_register->scriptHandle(),
+            'editor_style'    => $this->asset_register->styleHandle(),
+            'script'          => $this->asset_register->scriptHandle(),
+            'style'           => $this->asset_register->styleHandle(),
```

Comment:
Hmm In not sure about this.   `editor_script` and `script` _could_ be two different things because `editor_script` is intended for scripts only needed in the editor context and `script` is intended for scripts only needed in the frontend when the block is parsed for display (and the same with styles).  Now its entirely feasible that in most cases these scripts and styles will be the same in both contexts however  we should still allow for separation where needed.

Reply:
kk agreed... will add specific getters and setters